includes/db/DatabaseError.php

```diff
@@ -17,16 +17,22 @@ class DBError extends MWException {
 	 * @param $error String A simple error message to be used for debugging
 	 */
 	function __construct( DatabaseBase &$db, $error ) {
+		global $wgDBcluster;
+
 		$this->db = $db;
 		parent::__construct( $error );
 
+		$isMaster = !is_null( $db->getLBInfo( 'master' ) );
```

Comment:
We'll need to remember that master node can also be returned when you ask for a slave node depending on how weights are configured. Would be cool to have this information as well but I'm unsure it's possible with the current architecture. What do you think?